src/client/sandbox/timers.js

```diff
@@ -0,0 +1,112 @@
+    constructor () {
+        super();
+
+        this.timeouts          = [];
+        this.deferredFunctions = [];
+        this.setTimeout        = nativeMethods.setTimeout;
+    }
+
+    _wrapTimeoutFunctionsArguments (args) {
+        var isScriptFirstArg = typeof args[0] === 'string';
+        var func             = !isScriptFirstArg ? args[0] : null;
+        var script           = isScriptFirstArg ? processScript(args[0]) : null;
+
+        if (isIE && browserVersion < 12) {
+            var timersSandbox = this;
+            var fnToRun       = isScriptFirstArg ? () => this.window.eval(script) : func;
+
+            args[0] = function () {
+                return timersSandbox._callDeferredFunction(fnToRun, arguments);
+            };
+        }
+        else if (isScriptFirstArg)
+            args[0] = script;
+
+        return args;
+    }
+
+    _callDeferredFunction (fn, args) {
+        if (this.timeouts.length) {
+            var curTimeouts = [];
+            var curHandlers = [];
+
+            for (var i = 0; i < this.timeouts.length; i++) {
+                curTimeouts.push(this.timeouts[i]);
+                curHandlers.push(this.deferredFunctions[i]);
+            }
+
+            this.timeouts          = [];
+            this.deferredFunctions = [];
+
+            for (var j = 0; j < curTimeouts.length; j++) {
+                this.window.clearInterval(curTimeouts[j]);
+                curHandlers[j]();
+            }
+
+            // NOTE: Handlers can create new deferred functions.
+            return this._callDeferredFunction(fn, args);
+        }
+
+        return fn.apply(this.window, args);
+    }
+
+    attach (window) {
+        super.attach(window);
+
+        var timersSandbox = this;
+
+        window.setTimeout = function () {
+            // NOTE: shallow-copy the remaining args. Don't use arr.slice(), since it may leak the arguments object.
+            // See: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/arguments
+            var args = new Array(arguments.length);
+
+            for (var i = 0; i < args.length; ++i)
+                args[i] = arguments[i];
+
+            return nativeMethods.setTimeout.apply(window, timersSandbox._wrapTimeoutFunctionsArguments(args));
+        };
+
+        window.setInterval = function () {
+            // NOTE: shallow-copy the remaining args. Don't use arr.slice(), since it may leak the arguments object.
+            // See: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/arguments
+            var args = new Array(arguments.length);
+
+            for (var i = 0; i < args.length; ++i)
+                args[i] = arguments[i];
+
+            return nativeMethods.setInterval.apply(window, timersSandbox._wrapTimeoutFunctionsArguments(args));
+        };
+
+        // NOTE: We are saving setTimeout wrapper for internal using in case if the page-script replaces
+        // it with an invalid value.
```

Comment:
```
        // NOTE: We are saving the setTimeout wrapper for internal use in case the page-script replaces
        // it with an invalid value.
```